chain/client/validators.go

```diff
@@ -0,0 +1,65 @@
+package rpc
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+)
+
+type ValidatorMinimal struct {
+	Address     string `json:"address"`
+	VotingPower int64  `json:"voting_power"`
+}
+
+type ValidatorsMinimal struct {
+	BlockHeight int64              `json:"block_height"`
+	Validators  []ValidatorMinimal `json:"validators"`
+}
+
+func GetEVMValidators(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		node, err := cliCtx.GetNode()
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		validators, err := node.Validators(nil)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		validatorsOnETH := ValidatorsMinimal{}
+		validatorsOnETH.BlockHeight = validators.BlockHeight
+		validatorsOnETH.Validators = []ValidatorMinimal{}
+
+		for _, validator := range validators.Validators {
+			pubKeyBytes, ok := validator.PubKey.(secp256k1.PubKeySecp256k1)
+			if !ok {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, "fail to cast pubkey")
+				return
+			}
+
+			if pubkey, err := crypto.DecompressPubkey(pubKeyBytes[:]); err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			} else {
+				validatorsOnETH.Validators = append(
+					validatorsOnETH.Validators,
+					ValidatorMinimal{
+						Address:     fmt.Sprintf("0x%x", crypto.PubkeyToAddress(*pubkey)),
```

Comment:
Can we use `crypto.PubkeyToAddress(*pubkey).String()` instead?

Reply:
done